packages/styles/src/components/grid.scss

```diff
@@ -11,6 +11,95 @@
 @use './../variables/spacing';
 @use './../variables/grid';
 
+$container-padding-x-cache: 0;
+$container-fluid-padding-x-cache: 0;
+$gutter-x-cache: 0;
+
+.container {
+  padding-inline: var(--post-container-padding-x);
+}
+
+.container-fluid {
+  padding-inline: var(--post-container-fluid-padding-x);
+}
+
+@each $breakpoint in grid.$grid-breakpoints-list {
+  $container-padding-x: map.get(grid.$grid-container-padding, $breakpoint);
+  $container-fluid-padding-x: map.get(grid.$grid-container-fluid-padding, $breakpoint);
+  $gutter-x: map.get(grid.$grid-gutter-x, $breakpoint);
+
+  @if ($container-padding-x-cache != $container-padding-x) {
+    .container {
+      @if $breakpoint == 'xs' {
+        --post-container-padding-x: #{$container-padding-x};
+      } @else {
+        @include media-breakpoint-up($breakpoint) {
+          --post-container-padding-x: #{$container-padding-x};
+        }
+      }
+    }
+  }
+
+  @if ($container-fluid-padding-x-cache != $container-fluid-padding-x) {
+    .container-fluid {
+      @if $breakpoint == 'xs' {
+        --post-container-fluid-padding-x: #{$container-fluid-padding-x};
+      } @else {
+        @include media-breakpoint-up($breakpoint) {
+          --post-container-fluid-padding-x: #{$container-fluid-padding-x};
+        }
+      }
+    }
+  }
+
+  @if ($gutter-x-cache != $gutter-x) {
+    .row {
+      @if $breakpoint == 'xs' {
+        --bs-gutter-x: #{$gutter-x};
+      } @else {
+        @include media-breakpoint-up($breakpoint) {
+          --bs-gutter-x: #{$gutter-x};
+        }
+      }
+    }
+  }
+
+  $container-padding-x-cache: $container-padding-x;
+  $container-fluid-padding-x-cache: $container-fluid-padding-x;
+  $gutter-x-cache: $gutter-x;
+}
+
+// redefine gutter-utilities, to override the custom gutter-width above
+@each $infix in grid.$grid-breakpoints-list {
+  @each $postfix, $gutter in $gutters {
+    @if $infix == 'xs' {
+      .g-#{$postfix},
+      .gx-#{$postfix} {
+        --bs-gutter-x: #{$gutter};
+      }
+
+      // .g-#{$postfix},
+      // .gy-#{$postfix} {
+      //   --bs-gutter-y: #{$gutter};
+      // }
+    } @else {
+      @include media-breakpoint-up($infix) {
+        .g-#{$infix}-#{$postfix},
+        .gx-#{$infix}-#{$postfix} {
+          --bs-gutter-x: #{$gutter};
+        }
+
+        // .g-#{$infix}-#{$postfix},
+        // .gy-#{$infix}-#{$postfix} {
+        //   --bs-gutter-y: #{$gutter};
+        // }
+      }
+    }
+  }
+}
+
+// legacy code
```

Comment:
Should it be deprecated/removed?

Reply:
Not yet, there is another ticket to do this: [#2041](https://github.com/swisspost/design-system/issues/2041).